src/controllers/upload-files-controller.ts

```diff
@@ -0,0 +1,69 @@
+import { Request, Response, NextFunction } from 'express';
+import multer, { StorageEngine } from 'multer';
+import path from 'path';
+
+export interface MulterOptions {
+  storage?: string;
+  limit?: number;
+  extensions: string[];
+}
+
+const getStorage = (options: MulterOptions): StorageEngine => {
+  let storageType: StorageEngine = multer.memoryStorage();
+
+  if (options.storage === 'disk') {
+    storageType = multer.diskStorage({
+      destination: (
+        _req: Request,
+        _file: Express.Multer.File,
+        cb: (error: Error | null, path: string) => void,
+      ): void => {
+        cb(null, 'uploads');
+      },
+      filename: (
+        _req: Request,
+        file: Express.Multer.File,
+        cb: (error: Error | null, fileName: string) => void,
+      ): void => {
+        cb(null, `${Date.now()}_${file.originalname}`);
+      },
+    });
+  }
+
+  return storageType;
+};
+
+const upload = async (options: MulterOptions, req: Request, res: Response, next: NextFunction): Promise<void> => {
```

Comment:
I have my doubts about this kind of signature for the method. Will it work? At least `req`, `res` and next seems here redundant/

For example, I will make the following connection for middleware:

```js
router.post('/route', upload({options}), uploadPhoto);
```

I think it can be written in more simple manner

```js
const staticOptions = {};
const upload = options => {
   const mergedOptions = Object.assign({}, options, staticOptions);
   if (mergedOptions.any) {
      return multer(mergedOptions).any()
   }
   return multer(mergedOptions).none();
}
```